src/SDKs/Batch/DataPlane/Azure.Batch.IntegrationTests/InboundEndpointIntegrationTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace BatchClientIntegrationTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using BatchTestCommon;
+    using Fixtures;
+    using Microsoft.Azure.Batch;
+    using Microsoft.Azure.Batch.Common;
+    using IntegrationTestUtilities;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class InboundEndpointIntegrationTests
+    {
+        private readonly ITestOutputHelper testOutputHelper;
+        private readonly PoolFixture poolFixture;
+        private static readonly TimeSpan TestTimeout = TimeSpan.FromMinutes(3);
+
+        public InboundEndpointIntegrationTests(ITestOutputHelper testOutputHelper)
+        {
+            this.testOutputHelper = testOutputHelper;
+        }
+
+        [Fact]
+        [Trait(TestTraits.Duration.TraitName, TestTraits.Duration.Values.ShortDuration)]
+        public void WhenPoolCreatedWithInboundEndpoints_EndpointsAreReturnedByPoolAndComputeNodes()
+        {
+            Action test = () =>
```

Comment:
Often we write tests sync since that covers a bit more code (since the sync calls the async internally).